packages/yew/src/html/component/lifecycle.rs

```diff
@@ -345,22 +345,16 @@ impl Runnable for PropsUpdateRunner {
                     #[cfg(feature = "csr")]
                     ComponentRenderState::Render {
                         next_sibling: ref mut current_next_sibling,
-                        ref parent,
-                        ref bundle,
                         ..
                     } => {
-                        bundle.shift(parent, next_sibling.clone());
```

Comment:
Shifting here prevents a different issue.

When a `PropsUpdateRunning` is run, it may update the next_sibling information, this information, however, is not passed to the child components.
When a child component runs a render without the parent being re-rendered, it will appear in the outdated place.

Instead of removing it here, Yew should check whether `BTag` already has the same next sibling and stop shifting when that is the case.

Reply:
My previous comment has clearly indicated that it fixes a completely **different issue**.
Please read the previous comment carefully.

Removing shifting here causes a regression:

<img width="471" alt="pr-2705" src="https://user-images.githubusercontent.com/11693215/170687895-4da2fd20-5302-44c2-954a-7db5457dfebc.png">

Reproducible Example: 

https://github.com/futursolo/yew-pr-2705-regression

Reply:
I did not quite understand the scenario you were referring to, so the reproducing example is appreciated and will be made into a test case.